integrations/va-auth0/src/index.tsx

```diff
@@ -213,78 +252,116 @@ const handleFetchEvent: FetchEventCallback<Auth0RuntimeContext> = async (request
         router.get('/visitor-auth/response', async (request) => {
             if ('site' in siteInstallation && siteInstallation.site) {
                 const publishedContentUrls = await getPublishedContentUrls(context);
-                const privateKey = context.environment.signingSecrets.siteInstallation!;
-                let token;
-                try {
-                    token = await sign(
-                        { exp: Math.floor(Date.now() / 1000) + 1 * (60 * 60) },
-                        privateKey,
-                    );
-                } catch (e) {
-                    return new Response('Error: Could not sign JWT token', {
-                        status: 500,
-                    });
-                }
 
+                const includeClaimsInToken =
+                    siteInstallation?.configuration.include_claims_in_va_token;
                 const issuerBaseUrl = siteInstallation?.configuration.issuer_base_url;
                 const clientId = siteInstallation?.configuration.client_id;
                 const clientSecret = siteInstallation?.configuration.client_secret;
-                if (clientId && clientSecret) {
-                    const searchParams = new URLSearchParams({
-                        grant_type: 'authorization_code',
-                        client_id: clientId,
-                        client_secret: clientSecret,
-                        code: `${request.query.code}`,
-                        redirect_uri: `${installationURL}/visitor-auth/response`,
+
+                if (!clientId || !clientSecret || !issuerBaseUrl) {
+                    return new Response(
+                        'Error: Either client id, client secret or issuer base url is missing',
+                        {
+                            status: 400,
+                        },
+                    );
+                }
+
+                const searchParams = new URLSearchParams({
+                    grant_type: 'authorization_code',
+                    client_id: clientId,
+                    client_secret: clientSecret,
+                    code: `${request.query.code}`,
+                    redirect_uri: `${installationURL}/visitor-auth/response`,
+                });
+                const accessTokenURL = `${issuerBaseUrl}/oauth/token/`;
+                const auth0TokenResp = await fetch(accessTokenURL, {
+                    method: 'POST',
+                    headers: { 'content-type': 'application/x-www-form-urlencoded' },
+                    body: searchParams,
+                });
+
+                if (!auth0TokenResp.ok) {
+                    const errorResponse = await auth0TokenResp.json<Auth0TokenResponseError>();
```

Comment:
This line could crash. Should be optional, you should do another try / catch around that.

Reply:
Extending the condition to be more defensive.